conformance/lang/expressions/transactional-expr/transactional_expr.balt

```diff
@@ -0,0 +1,571 @@
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: check, commit-action, error, optional-type, transaction-stmt, transactional-expr
+
+function init() returns error? {
+    io:println(transactional); // output false
+    transaction {
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+    io:println(transactional); // output false
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: check, commit-action, error, function-call-expr, optional-type, transaction-stmt, transactional-expr,
+        transactional-function
+
+function init() returns error? {
+    transaction {
+        f1(); // output true
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+transactional function f1() {
+    io:println(transactional);
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: check, commit-action, error, function-call-expr, optional-type, transaction-stmt, transactional-expr,
+        transactional-function
+
+function init() returns error? {
+    transactional function () trxFunc = transactional function() {
+        io:println(transactional);
+    };
+
+    transaction {
+        trxFunc(); // output true
+        io:println(transactional); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: check, commit-action, error, explicit-anonymous-function-expr, isolated-function, optional-type,
+        transaction-stmt, transaction:Info, transaction:onCommit, transactional-expr, var
+
+function init() returns error? {
+    var onCommitFunc = isolated function(transaction:Info? info) {
+        io:println(transactional);
+    };
+
+    transaction {
+        transaction:onCommit(onCommitFunc); // output true
```

Comment:
Is the output validation line is correct. `onCommitFunc` will be executed when `commit` statement is reached? Hence, let's add it at `commit`